Mathlib/Order/Filter/Germ.lean

```diff
@@ -239,76 +238,77 @@ theorem coe_compTendsto' (f : α → β) {lc : Filter γ} {g : Germ lc α} (hg :
   rfl
 #align filter.germ.coe_comp_tendsto' Filter.Germ.coe_compTendsto'
 
-/-- Given a germ `f : germ l β` and a function `g : γ → α`, where `l : filter α`, if `g` tends
-to `l` along `lc : filter γ`, then the composition `f ∘ g` is well-defined as a germ at `lc`. -/
+/-- Given a germ `f : Germ l β` and a function `g : γ → α`, where `l : Filter α`, if `g` tends
+to `l` along `lc : Filter γ`, then the composition `f ∘ g` is well-defined as a germ at `lc`. -/
 def compTendsto (f : Germ l β) {lc : Filter γ} (g : γ → α) (hg : Tendsto g lc l) : Germ lc β :=
   f.compTendsto' _ hg.germ_tendsto
 #align filter.germ.comp_tendsto Filter.Germ.compTendsto
 
 @[simp]
 theorem coe_compTendsto (f : α → β) {lc : Filter γ} {g : γ → α} (hg : Tendsto g lc l) :
-    (f : Germ l β).comp_tendsto g hg = f ∘ g :=
+    (f : Germ l β).compTendsto g hg = f ∘ g :=
   rfl
 #align filter.germ.coe_comp_tendsto Filter.Germ.coe_compTendsto
 
-@[simp]
+-- @[simp] -- Porting note: simp cannot prove this
 theorem compTendsto'_coe (f : Germ l β) {lc : Filter γ} {g : γ → α} (hg : Tendsto g lc l) :
-    f.compTendsto' _ hg.germ_tendsto = f.comp_tendsto g hg :=
+    f.compTendsto' _ hg.germ_tendsto = f.compTendsto g hg :=
   rfl
 #align filter.germ.comp_tendsto'_coe Filter.Germ.compTendsto'_coe
 
-@[simp, norm_cast]
-theorem const_inj [NeBot l] {a b : β} : (↑a : Germ l β) = ↑b ↔ a = b :=
-  coe_eq.trans <| const_eventually_eq
+-- Porting note: used `const` instead of lift
+@[simp] -- Porting note: removed norm_cast
+theorem const_inj [NeBot l] {a b : β} : const l a = const l b ↔ a = b :=
```

Comment:
Why is this needed? Can't we make the coercion work?
I don't know how many proofs in downstream file make use of `norm_cast`... but they would all break.

Reply:
I think this is an alternative of `has_lift_t`, but as you say, we had better protect proofs using `norm_cast`. I'll replace `const` to a coercion and re-add `norm_cast`.